src/components/lab/Lab.js

```diff
@@ -128,53 +165,29 @@ class Lab extends Component {
           height={stageHeight}
         >
           <Layer>
-            <EmittedLine
-              charge={charge}
-              chargeOscillation={chargeOscillation}
-              direction={[0, DIRECTION_VALUE]}
-            />
-            <EmittedLine
-              charge={charge}
-              chargeOscillation={chargeOscillation}
-              direction={[0, -DIRECTION_VALUE]}
-            />
-            <EmittedLine
-              charge={charge}
-              chargeOscillation={chargeOscillation}
-              direction={[-DIRECTION_VALUE, 0]}
-            />
-            <EmittedLine
-              charge={charge}
-              chargeOscillation={chargeOscillation}
-              direction={[DIRECTION_VALUE, 0]}
-            />
-            <EmittedLine
-              charge={charge}
-              chargeOscillation={chargeOscillation}
-              direction={[DIRECTION_VALUE, -DIRECTION_VALUE]}
-            />
-            <EmittedLine
-              charge={charge}
-              chargeOscillation={chargeOscillation}
-              direction={[-DIRECTION_VALUE, DIRECTION_VALUE]}
-            />
-            <EmittedLine
-              charge={charge}
-              chargeOscillation={chargeOscillation}
-              direction={[-DIRECTION_VALUE, -DIRECTION_VALUE]}
-            />
-            <EmittedLine
-              charge={charge}
-              chargeOscillation={chargeOscillation}
-              direction={[DIRECTION_VALUE, DIRECTION_VALUE]}
-            />
+            {generateAngles(parseInt(numberOfLines, 10)).map((angle) => (
```

Comment:
You can `parseInt` in the `mapStateToProps` function or in the `reducer`, in order to avoid doing it on each `render`.